omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptStateConversionController2_0Test.java

```diff
@@ -0,0 +1,124 @@
+		conceptStateConversion.setProgramWorkflow(workflow);
+		conceptStateConversion.setProgramWorkflowState(state);
+
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+		uuid = conceptStateConversion.getUuid();
+	}
+
+	@Override
+	public String getURI() {
+		return "conceptstateconversion";
+	}
+
+	@Override
+	public String getUuid() {
+		return uuid;
+	}
+
+	@Override
+	public long getAllCount() {
+		return 2;
+	}
+
+	@Test
+	public void shouldCreateStateConversion() throws Exception {
+		Concept concept = Context.getConceptService().getConceptByUuid("0955b484-b364-43dd-909b-1fa3655eaad2");
+		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(RestTestConstants1_8.WORKFLOW_UUID);
+
+		ProgramWorkflowState state = new ProgramWorkflowState();
+		state.setConcept(concept);
+		state.setInitial(true);
+		state.setTerminal(false);
+
+		workflow.addState(state);
+		Context.getProgramWorkflowService().saveProgram(workflow.getProgram());
+
+		Context.flushSession();
+
+		String json =
+				"{\"concept\": \"" + concept.getUuid() + "\",\"programWorkflow\": \"" + workflow.getUuid()
+						+ "\",\"programWorkflowState\": \"" + state.getUuid() + "\"}";
+
+		SimpleObject newStateConversion = deserialize(handle(newPostRequest(getURI(), json)));
+
+		assertNotNull(newStateConversion);
+		String uuid = newStateConversion.get("uuid");
+
+		ConceptStateConversion createdStateConversion = Context.getProgramWorkflowService().getConceptStateConversionByUuid(uuid);
+		assertEquals(workflow.getUuid(), createdStateConversion.getProgramWorkflow().getUuid());
+		assertEquals(state.getUuid(), createdStateConversion.getProgramWorkflowState().getUuid());
+		assertEquals(concept.getUuid(), createdStateConversion.getProgramWorkflow().getConcept().getUuid());
+	}
+
+	@Test
+	public void shouldPurgeStateConversion() throws Exception {
+		Concept concept = Context.getConceptService().getConceptByUuid("0955b484-b364-43dd-909b-1fa3655eaad2");
+		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(RestTestConstants1_8.WORKFLOW_UUID);
+
+		ProgramWorkflowState state = new ProgramWorkflowState();
+		state.setConcept(concept);
+		state.setInitial(true);
+		state.setTerminal(false);
+
+		workflow.addState(state);
+		Context.getProgramWorkflowService().saveProgram(workflow.getProgram());
+		Context.flushSession();
+
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion();
+		conceptStateConversion.setConcept(concept);
+		conceptStateConversion.setProgramWorkflow(workflow);
+		conceptStateConversion.setProgramWorkflowState(state);
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+
+		handle(newDeleteRequest(getURI() + "/" + conceptStateConversion.getUuid(), new Parameter("purge", "true")));
+
+		assertNull(Context.getProgramWorkflowService().getConceptStateConversionByUuid(conceptStateConversion.getUuid()));
```

Comment:
How do i tell that the above was not null before the delete request?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/493/commits/3c9fee372101921ec32a3d3103b035e190ec4a33